jsonschema/cli.py

```diff
@@ -9,39 +13,134 @@
 from jsonschema import __version__
 from jsonschema._reflect import namedAny
 from jsonschema.validators import validator_for
+from jsonschema.exceptions import SchemaError, ValidationError
+
+
+# ------------------------------------------------------------------------------
+# CLASSES
+# ------------------------------------------------------------------------------
+class CliOutputWriter():
```

Comment:
For better or **worse** this module is still public (which #600 will fix) -- so we should be careful not to add any *more* public things to it. Can you rename this to be private?

But also -- I think you have the right idea here of an output writer, but if you use object polymorphism, things get even nicer.

I.e., rather than have every method have `if whatever == "pretty": do_something_pretty()`, have `_PrettyOutputWriter` and `_PlainOutputWriter` with the same interface, and then just select which writer is used based on the command line flag.

Lemme know if that makes sense to you, otherwise happy to elaborate.

Reply:
Of course that is much better ! I just didn’t think about it. But it is definitely the kind of things I like to write in Python.